packages/react/src/modules/slack/components/ConnectToSlackButton/ConnectToSlackButton.tsx

```diff
@@ -18,32 +16,70 @@ export const ConnectToSlackButton: React.FC<Props> = ({
   slackClientId,
   redirectUrl,
 }) => {
-  const knock = useKnockClient();
+  const { connectionStatus, actionLabel, errorLabel, setActionLabel, buildSlackAuthUrl, disconnectFromSlack } =
+    useManageSlackConnection(
+      knockSlackChannelId,
+      tenant,
+      slackClientId,
+      redirectUrl,
+    );
 
-  const rawParams = {
-    state: JSON.stringify({
-      redirect_url: redirectUrl,
-      access_token_object: {
-        object_id: tenant,
-        collection: TENANT_COLLECTION,
-      },
-      channel_id: knockSlackChannelId,
-      public_key: knock.apiKey,
-      user_token: knock.userToken,
-    }),
-    client_id: slackClientId,
-    scope: "chat:write,chat:write.public,channels:read,groups:read",
-  } as Record<string, string>;
-  const params = new URLSearchParams(rawParams);
+  // Loading states
+  if (connectionStatus === "loading" || connectionStatus === "disconnecting") {
```

Comment:
Nit: would consider moving to separate components but not necessarily something we need to do right now